### PR TITLE
Implement resource serialization for stdoutlog

### LINKED ISSUE #5211: stdoutlog: Implement Resource serialization

### PR DESCRIPTION
**Pull request by XSAM**
resolves #5211

Adding tests for this behavior is not easy until a test record structure like #5200 is implemented. I will include the test on #5212.

### CODE REVIEW

**Review by pellared — COMMENTED** [review]
Tests are failing

File: exporters/stdout/stdoutlog/exporter_test.go [hunk 1]
```diff
@@ -219,7 +219,7 @@ func getPrettyJSON(now time.Time) string {
 	"TraceID": "0102030405060708090a0b0c0d0e0f10",
 	"SpanID": "0102030405060708",
 	"TraceFlags": "01",
-	"Resource": {},
+	"Resource": null,
```
Comment (MrAlias) [inline exporters/stdout/stdoutlog/exporter_test.go:222]: Can we add a test that includes a serialized resource?

Comment (XSAM) [reply exporters/stdout/stdoutlog/exporter_test.go:222]: IIUC, there is no way to set the resource outside the `sdk/log` package.

**Review by pellared — APPROVED** [review]

**Review by MrAlias — APPROVED** [review]

**Comment by pellared** [comment]
@open-telemetry/go-approvers This needs a review from a non-Splunker.

**Review by dmathieu — APPROVED** [review]

**Pull request merged**